rfd/xxxx-automatic-azure-server-discovery.md

```diff
@@ -0,0 +1,105 @@
+---
+authors: Andrew Burke (andrew.burke@goteleport.com)
+state: draft
+---
+
+# RFD X - Automatic discovery of Azure servers
+
+## What
+
+Teleport SSH services will be able to automatically discover Azure virtual machine
+instances. See [RFD 57](https://github.com/gravitational/teleport/blob/master/rfd/0057-automatic-aws-server-discovery.md)
+(Automatic discovery and enrollment of AWS servers) for the same feature implemented
+for AWS.
+
+Note: unlike RFD 57, this RFD covers discovery only. Azure node joining will be added
+later.
+
+## Why
+
+RFD 57 replaced the process of manually installing Teleport on EC2 servers, which
+could be slow for large numbers of instances. This RFD replaces the process on Azure
+as well.
+
+## Discovery
+
+Azure discovery will be handled by the new [discovery service](https://github.com/gravitational/teleport/blob/master/rfd/0057-automatic-aws-server-discovery.md#discovery) described in RFD 57.
+
+```yaml
+discovery_service:
+  enabled: "yes"
+  azure:
```

Comment:
Is it legal to specify both `azure` and `aws` in the config for a single discovery service, or must you pick only one?

Reply:
The AKS and EKS discovery already uses both at the same time but the service must have authentication credentials for both.